src/mpid/ch4/netmod/ofi/ofi_pre.h

```diff
@@ -113,7 +113,12 @@ typedef struct {
     MPIDI_OFI_am_header_t msg_hdr;
     uint8_t am_hdr_buf[MPIDI_OFI_MAX_AM_HDR_SIZE];
     /* FI_ASYNC_IOV requires an iov storage to be alive until a request completes */
+#if MPIDI_OFI_IOVEC_ALIGN <= sizeof(void *)
```

Comment:
This doesn't work on Mac OS:

```
  CC       src/mpi/attr/lib_libmpi_la-attr_put.lo
In file included from ./src/mpid/ch4/include/netmodpre.h:15,
                 from ./src/mpid/ch4/include/mpidpre.h:25,
                 from ./src/include/mpiimpl.h:211,
                 from src/env/mpichversion.c:7:
./src/mpid/ch4/include/../netmod/ofi/ofi_pre.h:116:30: warning: "sizeof" is not defined, evaluates to 0 [-Wundef]
 #if MPIDI_OFI_IOVEC_ALIGN <= sizeof(void *)
                              ^~~~~~
./src/mpid/ch4/include/../netmod/ofi/ofi_pre.h:116:36: error: missing binary operator before token "("
 #if MPIDI_OFI_IOVEC_ALIGN <= sizeof(void *)
                                    ^
```

Reply:
Maybe something like this will work?

https://stackoverflow.com/a/51617465/491687

Reply:
It shouldn't work and I wonder why the tests still passes -- as if the Jenkins didn't wipe the workspace (even when it claim so). I'll double check that before update the fix.

Relying on `_Static_assert` will be less idea as it is a C11 thing. I am thinking that I can add a config macro for it. We already have size macros for the basic data types already.

Reply:
We have `SIZEOF_VOID_P` defined in `mpichconf.h`, so no problem here :)